### ISSUE TITLE
Difference between SalesforcePy and simple-salesforce

### ISSUE THREAD

**Issue by kokes** [forcedotcom/SalesforcePy#27]
Hi,
why are there two packages for interaction with the Salesforce API via Python? Is there a major difference between the two? Should we invest into one or the other? Where should we send pull requests?

The other project, which has been around for more than 5 years, has seen some traction already, so I'd like to know why this one exists. Especially given that [simple-salesforce](https://pypi.org/project/simple-salesforce/) has an SFDC employee listed as a maintainer.

It'd be great if any response to this issue could be included in this project's README.

Thanks!

**Comment by kokes** [comment]
cc @demianbrecht, perhaps you can shed some light on this, since you're the maintainer of simple-salesforce and a committer here. Thanks!

**Comment by demianbrecht** [comment]
Sorry for the delayed response @kokes, unfortunately I just caught this now.

TL;DR: Totally up to you, it's worth checking on functionality offered by each.

simple-salesforce isn't an official Salesforce project. SalesforcePy is maintained by a team at Salesforce.

At the end of the day, it all depends on what you're after I guess. I don't have a feature matrix readily available, but there are definitely differences. For example, SalesforcePy has a Chatter implementation while simple-salesforce doesn't. OTOH, simple-salesforce implements multiple methods of authentication via SOAP while SalesfocePy implement OAuth 2.0 [client credentials](https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/) flow.

From an architectural standpoint, SalesforcePy breaks service implementations up into submodules and exposes them through client attributes (i.e. `Client.chatter`) while simple-salesforce exposes everything through methods right from the client root. It's entirely personal preference of course, but I prefer the former and was intending on doing something similar with simple-salesforce before SalesforcePy came along and my use of simple-salesforce was removed from my day-to-day work.

**Comment by pepefloyd** [comment]
I think @demianbrecht summarized it very well.  There is a group of internal and external people behind this package which use it in their daily jobs so you can be sure it will continue to be maintained in the foreseeable future, is up to you which one you want to use.
